core/API/ResponseBuilder.php

```diff
@@ -137,7 +137,9 @@ public function getResponseException($e)
         $e       = $this->decorateExceptionWithDebugTrace($e);
         $message = $this->formatExceptionMessage($e);
 
-        if (
+        if (strpos($message, 'The method') !== false) {
```

Comment:
Is there possibly a better way to check if the message says that a method doesn't exist? Maybe a regular expression that would match the whole message with a wildcard for the method name?

Reply:
That actually can't work this way. The message of that exception is actually translatable. So depending on your language it might not even contain the string `The Method` or whatever. 
The correct way to solve that, would be throwing a HttpCodeException here instead:
https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/core/API/Proxy.php#L615-L620